src/Resources/config/services.xml

```diff
@@ -5,7 +5,8 @@
            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
 
     <services>
-        <defaults autowire="true"/>
+        <defaults autowire="true" autoconfigure="true"/>
+        <prototype namespace="Frosh\MailArchive\" resource="../../" />
 
         <service id="Frosh\MailArchive\Content\MailArchive\MailArchiveDefinition">
             <tag name="shopware.entity.definition" entity="frosh_mail_archive"/>
```

Comment:
shopware.entity.definition and shopware.entity.extension could be autowired aswell

Reply:
Thank you! 👍